v2/tools/generator/internal/config/type_configuration.go

```diff
@@ -26,13 +26,16 @@ import (
 // └──────────────────────────┘       └────────────────────┘       └──────────────────────┘       ╚═══════════════════╝       └───────────────────────┘
 //
 type TypeConfiguration struct {
-	name              string
-	properties        map[string]*PropertyConfiguration
-	nameInNextVersion configurableString
-	export            configurableBool
-	exportAs          configurableString
+	name                  string
+	properties            map[string]*PropertyConfiguration
+	nameInNextVersion     configurableString
+	export                configurableBool
+	exportAs              configurableString
+	azureGeneratedSecrets []*configurableString
```

Comment:
This is _kinda_ hacky. Let me know if you think I should make a `configurableStringSlice` instead.

Reply:
Whether this is hacky depends on the scope - if you might have _some_ of the string unconsumed, then the way you've done it is fine (and you'd generate an error for each one). But if it's all or nothing, a `configurableStringSlice` would be a better way to go.

Reply:
Also, I think you need to add `type_configuration_test.go` to make sure the unmarshal from yaml works correctly.

Reply:
It's either all consume or all unconsumed. Have moved this to a new `configurableStringSlice`.